pandas/tests/extension/base/ops.py

```diff
@@ -0,0 +1,89 @@
+import pytest
+
+import operator
+
+import pandas as pd
+from .base import BaseExtensionTests
+
+
+class BaseOpsUtil(BaseExtensionTests):
+    def check_opname(self, s, op_name, other, exc=NotImplementedError):
+
+        short_opname = op_name.strip('_')
+        try:
+            op = getattr(operator, short_opname)
+        except AttributeError:
+            # Assume it is the reverse operator
+            rop = getattr(operator, short_opname[1:])
+            op = lambda x, y: rop(y, x)
+
+        self._check_op(s, op, other, exc)
+
+    def _check_op(self, s, op, other, exc=NotImplementedError):
+        if exc is None:
+            result = op(s, other)
+            expected = s.combine(other, op)
+            self.assert_series_equal(result, expected)
+        else:
+            with pytest.raises(exc):
+                op(s, other)
+
+
+class BaseArithmeticOpsTests(BaseOpsUtil):
+    """Various Series and DataFrame arithmetic ops methods."""
+
+    def test_arith_scalar(self, data, all_arithmetic_operators):
+        # scalar
+        op_name = all_arithmetic_operators
+        s = pd.Series(data)
+        self.check_opname(s, op_name, s.iloc[0], exc=TypeError)
+
+    def test_arith_array(self, data, all_arithmetic_operators):
+        # ndarray & other series
+        op_name = all_arithmetic_operators
+        s = pd.Series(data)
+        self.check_opname(s, op_name, [s.iloc[0]] * len(s), exc=TypeError)
+
+    def test_divmod(self, data):
+        s = pd.Series(data)
+        self._check_op(s, divmod, 1, exc=TypeError)
+        self._check_op(1, divmod, s, exc=TypeError)
+
+    def test_error(self, data, all_arithmetic_operators):
+        # invalid ops
+        op_name = all_arithmetic_operators
+        with pytest.raises(AttributeError):
+            getattr(data, op_name)
+
+
+class BaseComparisonOpsTests(BaseOpsUtil):
+    """Various Series and DataFrame comparison ops methods."""
+
+    def _compare_other(self, data, op_name, other):
+        s = pd.Series(data)
+
+        if op_name == '__eq__':
```

Comment:
I think you should do here something similar as in the Arithmetic tests to convert the `__eq__` to `operator.eq` (so to not test `s.__eq__(other)` directly but to test `s == other`

Reply:
I did a mix.  We want to test that by default the attributes of the op_name are not there, but we still want to call the operators directly and see that they raise `TypeError` (except for eq and ne which are defined)